main/common/directives/urlified.js

```diff
@@ -22,9 +22,49 @@
       restrict: 'A',
       require: 'ngModel',
       link: function (scope, element, attrs, ngModelCtrl) {
-        scope.$watch(attrs.ngModel, function () {
+        scope.doGetCaretPosition = function doGetCaretPosition (element) {
```

Comment:
shouldn't it be named Text Cursor instead of Caret which is confusing with the small select box arrow?